app/gui/src/project-view/util/getIconName.ts

```diff
@@ -68,3 +72,21 @@ export function iconOfNode(node: NodeId, graphDb: GraphDb) {
       return 'data_input'
   }
 }
+
+/**
+ * Returns the icon to show on a component, using either the provided base icon or an icon representing its current
+ * status.
+ */
+export function useDisplayedIcon(
+  graphDb: GraphDb,
+  externalId: ToValue<ExternalId>,
+  baseIcon: ToValue<Icon | URLString>,
+) {
+  const evaluating = computed(() => {
+    const status = graphDb.getExpressionInfo(toValue(externalId))?.payload.type
+    return status === 'Pending' || status === undefined
+  })
```

Comment:
Task specs states, that we should display node as "evaluating" only when `payload.progress > 0`.

Engine does not send us progress [(yet)](https://github.com/enso-org/enso/issues/12009), but we may test it in integration tests (we already have `mockExpressionUpdate`)